src/transformers/models/mask2former/modeling_mask2former.py

```diff
@@ -2011,13 +2011,12 @@ def __init__(self, hidden_size: int, num_heads: int, mask_feature_size: torch.Te
     def forward(self, outputs: torch.Tensor, pixel_embeddings: torch.Tensor, attention_mask_target_size: int = None):
         mask_embeddings = self.mask_embedder(outputs.transpose(0, 1))
 
-        # Sum up over the channels
-        # (batch_size, num_queries, num_channels, 1, 1)
-        mask_embeddings = mask_embeddings.unsqueeze(-1).unsqueeze(-1)
-        # (batch_size, 1, num_channels, height, width)
-        pixel_embeddings = pixel_embeddings.unsqueeze(1)
-        # (batch_size, num_queries, height, width)
-        outputs_mask = (mask_embeddings * pixel_embeddings).sum(2)
+        # Equivalent to einsum('bqc, bchw -> bqhw') but jit friendly
+        batch_size, num_queries, num_channels = mask_embeddings.shape
+        _, _, height, width = pixel_embeddings.shape
+        outputs_mask = torch.zeros((batch_size, num_queries, height, width), device=mask_embeddings.device)
+        for c in range(num_channels):
+            outputs_mask += mask_embeddings[..., c][..., None, None] * pixel_embeddings[:, None, c]
```

Comment:
I have a bit concern about this per channel approach regarding the runtime efficiency, especially I think the `num_channels` is not just `3`.

Might be good to measure the timing required for this relevant lines: `einsum` v.s `per channel sum`.